### ISSUE TITLE
Analysis and cleanup jobs are inconsistently placed

### ISSUE THREAD

**Issue by achantavy** [cartography-cncf/cartography#820]
*Description*:
> What issue is being seen?  Describe what should be happening instead of the bug, for example: Cartography should not crash, the expected value isn't returned, the data schema is wrong, etc.

By default, the [final](https://github.com/lyft/cartography/blob/f28fcd7436c17d9cbf32a31161f1d592d1d29f2e/cartography/sync.py#L173) cartography sync stage runs all analysis jobs located in the cartography/data/jobs/analysis folder. I noticed that there are cases where we call `run_analysis_job()` out of band such as when syncing [iaminstanceprofiles](https://github.com/lyft/cartography/blob/f28fcd7436c17d9cbf32a31161f1d592d1d29f2e/cartography/intel/aws/__init__.py#L77-L81) and analyzing [lambda-to-ecr relationships](https://github.com/lyft/cartography/blob/f28fcd7436c17d9cbf32a31161f1d592d1d29f2e/cartography/intel/aws/__init__.py#L83-L87), in [S3 acls](https://github.com/lyft/cartography/blob/f28fcd7436c17d9cbf32a31161f1d592d1d29f2e/cartography/intel/aws/s3.py#L229-L234), everything [here](https://github.com/lyft/cartography/blob/f28fcd7436c17d9cbf32a31161f1d592d1d29f2e/cartography/intel/aws/__init__.py#L228-L244), and probably others.

The problem with these one-off calls of analysis jobs is that by default they will all be run a second time when we reach the final sync stage. This is wasted work and adds time to the sync especially on a large graph.

To summarize,

1. We need to run analysis on the entire graph. We currently do this in the default final sync stage where all jobs in cartography/data/jobs/analysis are run.
2. We need to run analysis on a segment of the graph, for example a single AWS account or single GCP project. We currently do this with one-off calls to `run_analysis_job()`.
3. We need to avoid running any of these jobs twice. This could be accomplished by splitting cartography/data/jobs/analysis into separate folders, or other ideas.

*Please complete the following information:*:
>  - Cartography release version or commit hash [e.g. 0.12.0 or 95e8e11913e2a44a4d4682506d8364a638ceac69]

0.56.0

**Comment by ramonpetgrave64** [comment]
From @ryan-lane https://github.com/lyft/cartography/pull/826#issuecomment-1109419468
> Could we separate out the analysis jobs that are intended to run along with
> the code, from the ones that are intended to connect nodes between modules?
> It may be good to put analysis and cleanup files for modules directly into
> their modules.
> 
> To be honest I wasn't aware that the folder was run at the end of every run
> and for anyone that changes the value of that option it's not how it runs.

**Comment by achantavy** [comment]
Closing, fixed in #853. There may be other issues, we can address later.
